elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/ElasticsearchHttpClientConfigurer.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.elasticsearch.client;
+
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+
+/**
+ * @author Yoann Rodiere
+ * @hsearch.experimental This contract is currently under active development and may be altered in future releases,
+ */
+public interface ElasticsearchHttpClientConfigurer {
+
+	HttpAsyncClientBuilder configure(HttpAsyncClientBuilder builder);
```

Comment:
Yes I understand what you wanted to do, but it might be confusing. I'd rather return void to make it clear what you expect people to do.

Methods which mutate parameters are not commonly used either but I think this is an example in which it's ok.